File: app/src/main/java/com/ihadzhi/eatlimination/ui/BaseFragment.java
package com.ihadzhi.eatlimination.ui;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.annotation.StringRes;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

public class BaseFragment extends Fragment {

    private ContainerInteractor containerInteractor;

    public void showBackButton() {
        ((AppCompatActivity) getActivity()).getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }

    public void hideBackButton() {
        ((AppCompatActivity) getActivity()).getSupportActionBar().setDisplayHomeAsUpEnabled(false);
    }

    public void setTitle(@StringRes int title) {
        ((AppCompatActivity) getActivity()).getSupportActionBar().setTitle(title);
    }

    public void setTitle(CharSequence title) {
        ((AppCompatActivity) getActivity()).getSupportActionBar().setTitle(title);
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        if (context instanceof ContainerInteractor) {
            containerInteractor = (ContainerInteractor) context;
        } else {
            throw new IllegalStateException("context should be instance of ContainerInteractor");
        }
    }

    public void showNavigation() {
        containerInteractor.showBottomNavigation();
    }

    public void hideNavigation() {
        containerInteractor.hideBottomNavigation();
    }

    public void showLoadingIndicator() {
        containerInteractor.showLoadingIndicator();
    }

    public void hideLoadingIndicator() {
        containerInteractor.hideLoadingIndicator();
    }
}

File: app/src/main/java/com/ihadzhi/eatlimination/ui/BaseFragment.kt
package com.ihadzhi.eatlimination.ui

import android.content.Context
import androidx.annotation.StringRes
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment

open class BaseFragment : Fragment() {
    private var containerInteractor: ContainerInteractor? = null
    fun showBackButton() {
        (activity as AppCompatActivity?)?.supportActionBar?.setDisplayHomeAsUpEnabled(true)
    }

    fun hideBackButton() {
        (activity as AppCompatActivity?)?.supportActionBar?.setDisplayHomeAsUpEnabled(false)
    }

    fun setTitle(@StringRes title: Int) {
        (activity as AppCompatActivity?)?.supportActionBar?.setTitle(title)
    }

    fun setTitle(title: CharSequence?) {
        (activity as AppCompatActivity?)?.supportActionBar?.title = title
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        containerInteractor = if (context is ContainerInteractor) {
            context
        } else {
            throw IllegalStateException("context should be instance of ContainerInteractor")
        }
    }

    fun showNavigation() {
        containerInteractor?.showBottomNavigation()
    }

    fun hideNavigation() {
        containerInteractor?.hideBottomNavigation()
    }

    fun showLoadingIndicator() {
        containerInteractor?.showLoadingIndicator()
    }

    fun hideLoadingIndicator() {
        containerInteractor?.hideLoadingIndicator()
    }

    fun hideSoftKeyboard() {
        containerInteractor?.hideSoftKeyboard()
    }

    fun showGenericError() {
        containerInteractor?.errorGeneric()
    }

    fun showError(@StringRes errorMessage: Int) {
        containerInteractor?.error(errorMessage)
    }
}